syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.routeguide";
option java_outer_classname = "RouteGuideProto";
option objc_class_prefix = "RTG";


package kapa.tarracsh.app.server.digest;

service PublicDigest {
  rpc Quit(Empty) returns (Empty) {}
  rpc Diff(DiffRequest) returns (DiffResponse) {}
}

message Empty {}

message DiffRequest {
    string input = 1;
    bool dryRun = 2;
}

message FileDigestResult {
    string filename = 1;
    bool isNew = 2;
    bool isModified = 3;
    bool isSameDigest = 4;
    bool failed = 5;
}

message ClassDigestResult {
    string strongClassname = 1;
    bool isNew = 2;
    bool isModified = 3;
    bool isSameDigest = 4;
    bool failed = 5;
}

message DiffResponse {
    repeated FileDigestResult files = 1;
    repeated ClassDigestResult classfiles = 2;
}
